addons/hashes/test_parseJSON.sqf

```diff
@@ -0,0 +1,132 @@
+_result sort true;
+_expected = ["address","age","companyname","firstname","lastname","newsubscription","phonenumber"]; //all lower case
+TEST_OP(_result,isEqualTo,_expected,_fn);
+
+_result = _data getVariable "lastName";
+_expected = "Smith";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = _data getVariable "age";
+_expected = 25;
+TEST_OP(_result,==,_expected,_fn);
+
+_result = _data getVariable "address" getVariable "city";
+_expected = "New York";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = _data getVariable "phoneNumber" select 0 getVariable "type";
+_expected = "home";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = _data getVariable "phoneNumber" select 1 getVariable "type";
+_expected = "fax";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = _data getVariable "newSubscription";
+TEST_FALSE(_result,_fn);
+
+_result = _data getVariable "companyName";
+TEST_TRUE(isNull _result,_fn);
+
+// Hash syntax
+_data = [preprocessFile "\x\cba\addons\hashes\test_parseJSON_config.json", true] call CBA_fnc_parseJSON;
+
+_result = [_data] call CBA_fnc_isHash;
+TEST_TRUE(_result,_fn);
+
+_result = [_data] call CBA_fnc_hashKeys;
+_result sort true;
+_expected = ["address","age","companyName","firstName","lastName","newSubscription","phoneNumber"]; //camel case
+TEST_OP(_result,isEqualTo,_expected,_fn);
+
+_result = [_data, "lastName"] call CBA_fnc_hashGet;
+_expected = "Smith";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = [_data, "age"] call CBA_fnc_hashGet;
+_expected = 25;
+TEST_OP(_result,==,_expected,_fn);
+
+_result = [[_data, "address"] call CBA_fnc_hashGet, "city"] call CBA_fnc_hashGet;
+_expected = "New York";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = [[_data, "phoneNumber"] call CBA_fnc_hashGet select 0, "type"] call CBA_fnc_hashGet;
+_expected = "home";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = [[_data, "phoneNumber"] call CBA_fnc_hashGet select 1, "type"] call CBA_fnc_hashGet;
+_expected = "fax";
+TEST_OP(_result,==,_expected,_fn);
+
+_result = [_data, "newSubscription"] call CBA_fnc_hashGet;
+TEST_FALSE(_result,_fn);
+
+_result = [_data, "companyName"] call CBA_fnc_hashGet;
+TEST_TRUE(isNull _result,_fn);
+
+// ----------------------------------------------------------------------------
+
+_fn = "CBA_fnc_encodeJSON";
+LOG("Testing " + _fn);
+
+TEST_DEFINED("CBA_fnc_encodeJSON",_fn);
+
+// ----------------------------------------------------------------------------
+
+private _testCases = [
+    "null",
+    "true",
+    "1.2",
+    """Hello, World!""",
+    "[]",
+    "{}",
+    "[null, true, 1.2, ""Hello, World!""]",
+    "[{""nested"": [{""nested"": [{""nested"": [{""nested"": [{""nested"": [{""nested"": [{""nested"": []}]}]}]}]}]}]}]"
+];
+
+{
+    private _useHashes = _x;
+
+    {
+        diag_log _x;
+        private _input  = _x;
+        private _object = [_x, _useHashes] call CBA_fnc_parseJSON;
+        private _output = [_object] call CBA_fnc_encodeJSON;
+        TEST_OP(_input,==,_output,_fn);
```

Comment:
`TEST_OP` is logging asynchronous, `diag_log` at command execution. This makes the logging appear in RPT out of order.

Either use `LOG_1(_x)` (requires `#define DEBUG_MODE_FULL` in L1), or `#define DEBUG_SYNCHRONOUS` in L1.